src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -517,6 +517,23 @@ public T WithTelemetry(ITelemetryConfig telemetryConfig)
             return (T)this;
         }
 
+        /// <summary>
+        /// Sets telemetry client for the application.
+        /// </summary>
+        /// <param name="telemetryClients">List of telemetry clients to add telemetry logs.</param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public T WithTelemetryClient(params ITelemetryClient[] telemetryClients)
```

Comment:
Do we expect multiple implementations of clients? How does SAL implement this? The spec doesn't specify either.

Reply:
The idea behind multiple clients as an extensibility model is that say substrate or teams can pass their own ITelemetryClient implementation that can decide which events it cares about and send them back to whatever backend they want. They might decide to take just one event type and push that to their Geneva tenant. That won't interfere with the common telemetry client we implement for 1P partners which emits to our shared store.